packages/fs-components/src/layouts/NavigationLayout.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { cloneElement, ComponentProps, ReactElement, Component } from "react";
+import { PropsWithChildren, useState } from "react";
+import NavigationBar from "../components/NavigationBar";
+
+// Do this: https://stackoverflow.com/questions/41881912/typing-a-react-component-that-clones-its-children-to-add-extra-props-in-typescri
+
+type NavigationBarProps = ComponentProps<typeof NavigationBar>
+
+interface Props {
+    navigationBarProps: Omit<NavigationBarProps, "_onMenuClick">;
+} 
+
+function NavigationLayout({ navigationBarProps, children }: PropsWithChildren<Props>) {
+    const [hamburgerMenuVisible, setHamburgerMenuVisible] = useState(false);
```

Comment:
I don't think that the hamburger menu visibility controls should be in the layout.

Examples: 
- https://react.carbondesignsystem.com/?path=/story/components-ui-shell--header-base-w-navigation-actions-and-side-nav
- https://react-bootstrap.github.io/components/navbar/#api